borg/archiver.py

```diff
@@ -625,6 +625,16 @@ def do_rename(self, args, repository, manifest, key, cache, archive):
         return self.exit_code
 
     @with_repository(exclusive=True, cache=True)
+    @with_archive
+    def do_comment(self, args, repository, manifest, key, cache, archive):
+        """Set the archive comment"""
+        archive.set_comment(args.comment)
```

Comment:
I think it's better to have that implementation detail in the Archive class, like rename (<-- both share a good bit of code, maybe deduplicate that?).